src/components/TrackMetadataPopper.tsx

```diff
@@ -0,0 +1,319 @@
+/*!
+ * © 2023 Atypon Systems LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  CHANGE_OPERATION,
+  CHANGE_STATUS,
+  trackCommands,
+  TrackedAttrs,
+} from '@manuscripts/track-changes-plugin'
+import { Command } from 'prosemirror-state'
+import React, { useMemo } from 'react'
+import styled from 'styled-components'
+
+import { TextButton } from './AlertMessage'
+import { ButtonGroup } from './Button'
+import { EditButtonIcon } from './icons'
+
+const getLatest = (a: TrackedAttrs, b: TrackedAttrs) =>
+  a.updatedAt > b.updatedAt ? a : b
+
+export const TrackNodeAttrs: React.FC<{
+  dataTracked?: TrackedAttrs[]
+}> = ({ dataTracked, children }) => {
+  const currentChange = useMemo<TrackedAttrs | undefined>(
+    () => dataTracked?.reduce(getLatest),
+    [dataTracked]
+  )
+
+  if (!currentChange) {
+    return <>{children}</>
+  }
+
+  const markStyle = getMarkStyle(currentChange)
+
+  const showAttrsButton =
+    currentChange.operation === CHANGE_OPERATION.set_node_attributes &&
+    currentChange.status === CHANGE_STATUS.pending
+
+  return (
+    <Mark
+      className={(showAttrsButton && 'attrs-track-mark') || undefined}
+      {...markStyle}
+    >
+      {children}
+      {showAttrsButton && <TrackAttrsButton changeId={currentChange.id} />}
+    </Mark>
+  )
+}
+
+const Mark = styled.div<{
+  background?: string
+  textDecoration?: string
+  display?: string
+}>`
+  position: relative;
+  ${(props) => props.background && `background: ${props.background}`};
+  ${(props) =>
+    props.textDecoration && `text-decoration: ${props.textDecoration}`};
+  ${(props) => props.display && `display: ${props.display}`};
+
+  :has(.attrs-popper-button, .attrs-popper-button-active):hover {
+    background: #a3eaff;
+  }
+
+  :hover .attrs-popper-button,
+  .attrs-popper-button-active {
+    display: inline-flex;
+  }
+
+  .attrs-popper-button,
+  .attrs-popper-button-active {
+    position: absolute;
+    bottom: 0;
+    right: 0;
+  }
+`
+
+export const getMarkStyle = (dataTracked?: TrackedAttrs) => {
```

Comment:
Is it possible to have this return a class name, and have the actual styles defined in `Mark`?

Reply:
updated to return class name, but will keep the actual style so we could use it for both react component & prosemirror view